### PR TITLE
[S3] content md5 in PutObjectResult

### PR DESCRIPTION
**Pull request by raphkim**
*Issue #, if available:*
#1236 

*Description of changes:*
Added a line to set content MD5 in `PutObjectResult` to match the behavior in [aws-sdk-java](https://github.com/aws/aws-sdk-java/blob/5bafd0d6c4a979bc55fcf6709e93c9983bd2d1e6/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/AmazonS3Client.java#L5812).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-android-sdk-s3/src/main/java/com/amazonaws/services/s3/AmazonS3Client.java [hunk 1]
```diff
@@ -2003,6 +2003,7 @@ public PutObjectResult putObject(PutObjectRequest putObjectRequest)
         result.setETag(returnedMetadata.getETag());
         result.setMetadata(returnedMetadata);
         result.setRequesterCharged(returnedMetadata.isRequesterCharged());
+        result.setContentMd5(contentMd5);
```
Comment (kvasukib) [inline aws-android-sdk-s3/src/main/java/com/amazonaws/services/s3/AmazonS3Client.java]: can you add unit and/or integration test for contentMd5 to be null, non-null and contentMd5 passed through ObjectMetadata by the developer?
